extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/extractors/YoutubeSubscriptionExtractor.java

```diff
@@ -68,4 +97,69 @@ public String getRelatedUrl() {
         }
         return subscriptionItems;
     }
+
+    public List<SubscriptionItem> fromZipInputStream(@Nonnull final InputStream contentInputStream)
+            throws ExtractionException {
+        ZipInputStream zipInputStream = new ZipInputStream(contentInputStream);
+
+        try {
+            ZipEntry zipEntry;
+            while ((zipEntry = zipInputStream.getNextEntry()) != null) {
+                if (IsEntryPathMatch(zipEntry.getName())) {
+                    try {
+                        return fromCsvInputStream(zipInputStream);
+                    }
+                    catch (ExtractionException e) {
+                        throw new InvalidSourceException("Error reading contents of file '" + zipEntry.getName() + "'");
+                    }
+                }
+            }
+        } catch (IOException e) {
+            throw new InvalidSourceException("Error reading contents of zip file", e);
+        }
+
+        throw new InvalidSourceException("Unable to find a subscriptions.csv file (try extracting and selecting the csv file)");
+    }
+
+    public static final String[] ENTRY_PATHS = {
+            "Takeout/YouTube and YouTube Music/subscriptions/subscriptions.csv",
+            "Takeout/YouTube y YouTube Music/suscripciones/suscripciones.csv" // There is a <NBSP> between YouTube and Music
+    };
```

Comment:
You can't rely on a list of paths. It will be different for each language. I'd navigate the whole zip searching for a valid file instead.

Reply:
A possible way to find this file is to look for files in the glob `Takeout/YouTube*/*/*.csv` and find one where the file starts with a match for this regex (this regex is a JS one, unsure of the Java equiv):

`/^.*\n([0-9A-Za-z_-]{20,}),[^,]*?\1,.+\n/`

Reply:
^ intuition, this regex skips the first line, checks the second line to see if it starts with something that looks like a channel id, then looks for `/` followed by that same id again followed by a comma `,`

Reply:
Good idea, it should work out fine

Reply:
Done--brief summary of changes:

CSV: after 5 lines, checks to make sure it's found subscription items (ensures the 2nd entry is a channel url)
if there's no items, then it exits out of the function, returning an empty list.

ZIP: iterates over every CSV file, gets the subscription items from that CSV file, if there's an error or no items in the list, it goes to the next CSV file in the zip.